### PR TITLE
Added recommended code style

### PR DESCRIPTION
**Pull request by owaiskazi19**
Signed-off-by: Owais Kazi <owaiskazi19@gmail.com>

### Description
Added Spotless as a recommended formatting plugin.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: STANDARDS.md [hunk 1]
```diff
@@ -29,6 +29,19 @@ Make sure your plugin follows the SoC design principle and has different functio
 
 Make effective use of namespacing since it will allow co-existence of ambiguous names / classes with the same name and help avoid collisions.
```
Comment (saratvemulapalli) [inline STANDARDS.md:31]: Please update table of contents.

File: STANDARDS.md [hunk 2]
```diff
@@ -29,6 +29,19 @@ Make sure your plugin follows the SoC design principle and has different functio
...
 
+### Recommended Code Style
```
Comment (saratvemulapalli) [inline STANDARDS.md:32]: All these are recommendations anyway. 
```suggestion
### Code Styling
```

File: STANDARDS.md [hunk 3]
```diff
@@ -29,6 +29,19 @@ Make sure your plugin follows the SoC design principle and has different functio
...
+Format Java files with the Eclipse JDT formatter, using the [Spotless Gradle](https://github.com/diffplug/spotless/tree/master/plugin-gradle) plugin.
```
Comment (saratvemulapalli) [inline STANDARDS.md:33]: What is JDT?

File: STANDARDS.md [hunk 4]
```diff
@@ -29,6 +29,19 @@ Make sure your plugin follows the SoC design principle and has different functio
...
+
```
Comment (saratvemulapalli) [inline STANDARDS.md:34]: Could you link sample formatting rules file ?

File: STANDARDS.md (hunk 2 above)
Comment (saratvemulapalli) [inline STANDARDS.md:32]: Logically I would prefer this to be under #Plugin Standards instead of Development Practices.

File: STANDARDS.md [hunk 5]
```diff
@@ -29,6 +29,19 @@ Make sure your plugin follows the SoC design principle and has different functio
...
+* Wildcard imports (`import foo.bar.baz.*`) are forbidden and will cause the build to fail.
+* If *absolutely* necessary, you can disable formatting for regions of code with the `// tag::NAME` and `// end::NAME` directives, but note that these are intended for use in documentation, so please make it clear what you have done, and only do this where the benefit clearly outweighs the decrease in consistency.
+* Note that JavaDoc and block comments i.e. `/* ... */` are not formatted, but line comments i.e `// ...` are.
+* There is an implicit rule that negative boolean expressions should use the form `foo == false` instead of `!foo` for better readability of the code. While this isn't strictly enforced, if might get called out in PR reviews as something to change.
+
```
Comment (saratvemulapalli) [inline STANDARDS.md:44]: Some plugins use checkstyle and spotless? I know we dont use checkstyle anymore. 
Would you recommend similar setup for plugins?

File: STANDARDS.md (hunk 2 above)
Comment (owaiskazi19) [reply STANDARDS.md:32]: I can do that but shouldn’t formatting be under Development Practices?

File: STANDARDS.md (hunk 3 above)
Comment (owaiskazi19) [reply STANDARDS.md:33]: Java Development Tools. Will push the change

File: STANDARDS.md (hunk 5 above)
Comment (owaiskazi19) [reply STANDARDS.md:44]: Should I link the META issue here to explain the process of removal of checkstyle and enabling spotless?

Comment (saratvemulapalli) [reply STANDARDS.md:45]: Um no, my question was slightly different. 
Would you recommend plugins to use only (and why):
- Spotless
- Spotless and Checkstyle
- Checkstyle 
- etc

Comment (owaiskazi19) [reply STANDARDS.md:45]: Pushed the change. Let me know what you think of it. I'll be happy to make more changes based on the suggestion. 😁

Comment (saratvemulapalli) [reply STANDARDS.md:45]: Looks good to me. 
Probably put this section at the top(L43) L46-50 to help the readers understand why we recommend spotless for OpenSearch plugins.

Comment (owaiskazi19) [reply STANDARDS.md:51]: Sure. Done!

**Review by saratvemulapalli — APPROVED** [review]
Looks good to me.

Thanks @owaiskazi19 !! good work!

**Pull request merged**